ChegeBryan/black-bandana
# #163261740 Register user with data validations

#### What does this PR do?

User registration with validations

#### Description of Task to be completed?
 
- Write tests for user various validations used
- Write appropriate responses for the validations

#### How should this be manually tested?
After cloning the repo to your local machine - check [README.md](README.md) for instructions. checkout to the branch `ft-register-user-api-II-163261740` Using Postman open a `POST` request with this header
Key Content-Type: application/json
Enter the following in the request body in json 
```
{
    "user_name": "test username",
    "user_email": "user@emai.com",
    "password": "justanoter23"
}
```
Tweak the different properties with both valid and invalid data.

#### What are the relevant pivotal tracker stories?
[#163261740](https://www.pivotaltracker.com/story/show/163261740)

#### Postman Documentation
https://documenter.getpostman.com/view/4869820/RzthQWgJ

#### Screenshots (if appropriate)

#####  `POST /users`

![screenshot from 2019-01-26 18-19-24](https://user-images.githubusercontent.com/40359451/51789213-8f54fa00-2197-11e9-9f94-85907f4f608c.png)
